File: DavidHassen.DAL/ImageProvider.cs
using DavidHassen.Shared;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;

namespace DavidHassen.DAL
{
    public class ImageProvider: IImageProvider
    {

        /// <summary>
        /// Insert ImageLibrary information.
        /// </summary>
        /// <remarks>
        /// <param name="imageModel">Passing all ImageLibrary info."</param> 
        /// <returns>returning ImageLibraryId which is generated.</returns>
        /// </remarks>
        public bool Insert(ImageModel imageModel)
        {
            var conn = new SqlConnection(SqlHelper.GetConnectionString());
            SqlTransaction sqlTrans = null;
            var param = new List<SqlParameter>();
            try
            {
                conn.Open();
                sqlTrans = conn.BeginTransaction();
                /*Add the Parameters*/

                param.Add(new SqlParameter { ParameterName = "@ImageId", Value = imageModel.ImageId });
                param.Add(new SqlParameter { ParameterName = "@ImageName", Value = imageModel.ImageName });
                param.Add(new SqlParameter { ParameterName = "@CroppedImagePath", Value = imageModel.CroppedImagePath });
                param.Add(new SqlParameter { ParameterName = "@CroppedThumbnailPath", Value = imageModel.CroppedThumbnailPath });
                param.Add(new SqlParameter { ParameterName = "@OriginalImagePath", Value = imageModel.OriginalImagePath });
                param.Add(new SqlParameter { ParameterName = "@CreatedDate", Value = imageModel.CreatedDate });
                param.Add(new SqlParameter { ParameterName = "@UpdateDate", Value = imageModel.UpdateDate });

                SqlHelper.ExecuteScalar(sqlTrans, CommandType.StoredProcedure, "spInsertImage", param.ToArray());
                //Commit transaction for saving info
                sqlTrans.Commit();

                if (conn.State == ConnectionState.Open)
                    conn.Close();
                return true;
            }
            catch  //Error...
            {
                if (sqlTrans != null)
                    if (sqlTrans.Connection.State == ConnectionState.Open)
                    {
                        //rollback transaction 
                        sqlTrans.Rollback();
                        conn.Close();
                    }
            }
            finally
            {
                if (conn.State == ConnectionState.Open)
                {
                    //closing connection
                    conn.Close();
                }
            }
            return false;
        }


    }
}

File: Data/DavidHassen.DAL/ImageProvider.vb
Imports System.Data.SqlClient
Imports DavidHassen.Shared

Public Class ImageProvider
    Implements IImageProvider

    ''' <summary>
    ''' Insert ImageLibrary information.
    ''' </summary>
    ''' <remarks>
    ''' <param name="imageModel">Passing all ImageLibrary info."</param> 
    ''' <returns>returning ImageLibraryId which is generated.</returns>
    ''' </remarks>
    Public Function Insert(ByVal imageModel As ImageModel) As Boolean Implements IImageProvider.Insert
        Dim conn = New SqlConnection(SqlHelper.GetConnectionString)
        Dim sqlTrans As SqlTransaction = Nothing
        Dim param = New List(Of SqlParameter)
        Try
            conn.Open()
            sqlTrans = conn.BeginTransaction

            Dim n = New SqlParameter
            n.ParameterName = "@ImageName"
            n.Value = imageModel.ImageName
            param.Add(n)

            n = New SqlParameter
            n.ParameterName = "@CroppedImagePath"
            n.Value = imageModel.CroppedImagePath
            param.Add(n)

            n = New SqlParameter
            n.ParameterName = "@CroppedThumbnailPath"
            n.Value = imageModel.CroppedThumbnailPath
            param.Add(n)

            n = New SqlParameter
            n.ParameterName = "@OriginalImagePath"
            n.Value = imageModel.OriginalImagePath
            param.Add(n)

            n = New SqlParameter
            n.ParameterName = "@CreatedDate"
            n.Value = imageModel.CreatedDate
            param.Add(n)

            n = New SqlParameter
            n.ParameterName = "@UpdateDate"
            n.Value = imageModel.UpdateDate
            param.Add(n)

            SqlHelper.ExecuteScalar(sqlTrans, CommandType.StoredProcedure, "spInsertImage", param.ToArray)
            'Commit transaction for saving info
            sqlTrans.Commit()

            If (conn.State = ConnectionState.Open) Then
                conn.Close()
            End If

            Return True
        Catch ex As System.Exception
            If (Not (sqlTrans) Is Nothing) Then
                If (sqlTrans.Connection.State = ConnectionState.Open) Then
                    'rollback transaction 
                    sqlTrans.Rollback()
                    conn.Close()
                End If

            End If

        Finally
            If (conn.State = ConnectionState.Open) Then
                'closing connection
                conn.Close()
            End If

        End Try

        Return False
    End Function

End Class
